src/annotator/config/index.js

```diff
@@ -160,7 +164,12 @@ const configDefinitions = {
   showHighlights: {
     allowInBrowserExt: false,
     defaultValue: 'always',
-    getValue: settings => settings.showHighlights,
+    validate: validators.oneOfType([
+      validators.isString.oneOf(['always', 'never', 'whenSidebarOpen']),
+      validators.isBoolean, // allow booleans for backwards compatibility
+    ]),
+    coerce: toShowHighlights, // transforms booleans to strings
+    getValue: getHostPageSetting,
   },
```

Comment:
The above code is the important thing for this draft. Do we want to add an optional "validate" method. If we did , it could look something like this for `showHighlights`. I'm also sure we can also immediately find other uses for it for just about every single config value.

The nice thing about doing this is it really separates the logic for invalid values from the user from coerced values that the app needs internally. In other words, the user does not care about showHighlights being a string internally, all they need to know is it the value they provided is a problem or not.  This gives them that feedback.

This also alleviates the messiness of the coerce methods also acting as pseudo validators.  Follow up work would be to further simply the existing coerce methods because if we had validation in front of them, then we can assume very strict incoming param values.